mongodbatlas/resource_mongodbatlas_cluster.go

```
@@ -1273,8 +1275,14 @@ func flattenProviderSettings(d *schema.ResourceData, settings *matlas.ProviderSe
 		log.Printf(errorClusterSetting, "provider_name", clusterName, err)
 	}
 
-	if err := d.Set("provider_region_name", settings.RegionName); err != nil {
-		log.Printf(errorClusterSetting, "provider_region_name", clusterName, err)
+	if val, ok := d.GetOk("provider_region_name"); ok {
+		if err := d.Set("provider_region_name", val); err != nil {
+			log.Printf(errorClusterSetting, "provider_region_name", clusterName, err)
+		}
```

shouldn't we return an error here?